File: 2024/01A.py
import sys

def solve(input):
    nums1 = sorted([int(line.split()[0]) for line in input.splitlines()])
    nums2 = sorted([int(line.split()[1]) for line in input.splitlines()])

    result = 0
    for (num1, num2) in zip(nums1, nums2):
        result += abs(num1 - num2)

    return result

if __name__ == '__main__':
    print(solve(open(sys.argv[1]).read()))   

File: 2024/01A.js


function solve(input) {
  const l1 = input.split('\n').map((row) => parseInt(row.split('  ')[0])).sort();
  const l2 = input.split('\n').map((row) => parseInt(row.split('  ')[1])).sort();

  var result = 0;
  for (let i = 0; i < l1.length ; i++){
    result += Math.abs(l1[i] - l2[i])
  }

  return result;
}

console.log(solve(require('node:fs').readFileSync(process.argv[2], 'utf8').trim()));
